From: Darren Kenny <darren.kenny@oracle.com>
Date: Thu, 5 Nov 2020 10:15:25 +0000
Subject: kern/efi: Fix memory leak on failure

Free the memory allocated to name before returning on failure.

Fixes: CID 296222

Signed-off-by: Darren Kenny <darren.kenny@oracle.com>
Reviewed-by: Daniel Kiper <daniel.kiper@oracle.com>
---
 grub-core/kern/efi/efi.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/grub-core/kern/efi/efi.c b/grub-core/kern/efi/efi.c
index ff200a0..2e88507 100644
--- a/grub-core/kern/efi/efi.c
+++ b/grub-core/kern/efi/efi.c
@@ -393,6 +393,7 @@ grub_efi_get_filename (grub_efi_device_path_t *dp0)
 	    {
 	      grub_error (GRUB_ERR_OUT_OF_RANGE,
 			  "malformed EFI Device Path node has length=%d", len);
+	      grub_free (name);
 	      return NULL;
 	    }
 
